### PR TITLE
Update deprecated APIs for Deno v1.40

### PR DESCRIPTION
**Pull request by littledivy**

### CODE REVIEW

**Comment by littledivy** [comment]
We're still merging a few deprecations, marking as draft.

**Comment by magurotuna** [comment]
Is this replacing #1228?

**Comment by littledivy** [comment]
@magurotuna Yes

File: docs/rules/no_deprecated_deno_api.md [hunk 1]
```diff
@@ -108,3 +108,47 @@ function foo(file: Deno.FsFile) {
...
+
+Deno.resources() was deprecated. There are no replacements for this API.
+
+- `Deno.metrics()`
+
+Deno.metrics() was deprecated. There are no replacements for this API.
+
+**HTTP server API**
+
+- `Deno.serveHttp`
```
Comment (bartlomieju) [inline docs/rules/no_deprecated_deno_api.md:152]: Can you add examples for `Deno.close()`/Deno.write etc too?

File: docs/rules/no_deprecated_deno_api.md [hunk 2]
```diff
@@ -108,3 +108,47 @@ function foo(file: Deno.FsFile) {
...
+- `Deno.fdatasyncSync`
+
+`Deno.fdatasync` and `Deno.fdatasyncSync` are deprecated in favor of
+`Deno.FsFile.datasync` and `Deno.FsFile.datasyncSync` respectively.
+
+- `Deno.fsync`
+- `Deno.fsyncSync`
+
+`Deno.fsync` and `Deno.fsyncSync` are deprecated in favor of `Deno.FsFile.sync`
+and `Deno.FsFile.syncSync` respectively.
```
Comment (iuioiua) [inline docs/rules/no_deprecated_deno_api.md:122]:
```suggestion
`Deno.fsync` and `Deno.fsyncSync` are deprecated in favor of `Deno.FsFile.flushAll`
and `Deno.FsFile.flushAllSync`, respectively.
```
Ref https://github.com/denoland/deno/pull/22085

File: docs/rules/no_deprecated_deno_api.md [hunk 3]
```diff
@@ -108,3 +108,47 @@ function foo(file: Deno.FsFile) {
...
+
+- `Deno.isatty`
+
+`Deno.isatty` was deprecated in favor of `Deno.stdin.isTerminal()`.
```
Comment (iuioiua) [inline docs/rules/no_deprecated_deno_api.md:126]:
```suggestion
`Deno.isatty` was deprecated in favor of `Deno.stdin.isTerminal()`, `Deno.stdout.isTerminal()` and `Deno.stderr.isTerminal()`.
```

File: docs/rules/no_deprecated_deno_api.md [hunk 4]
```diff
@@ -108,3 +108,47 @@ function foo(file: Deno.FsFile) {
...
+```typescript
+Deno.isatty(Deno.stdin.rid);
+Deno.isatty(Deno.stdout.rid);
+```
+
+### Valid:
+
+```typescript
+Deno.stdin.isTerminal();
+Deno.stdout.isTerminal();
```
Comment (iuioiua) [inline docs/rules/no_deprecated_deno_api.md:139]:
```suggestion
Deno.stdout.isTerminal();
Deno.stderr.isTerminal();
```

File: docs/rules/no_deprecated_deno_api.md [hunk 5]
```diff
@@ -108,3 +108,47 @@ function foo(file: Deno.FsFile) {
...
+
+- `Deno.isatty`
+
+`Deno.isatty` was deprecated in favor of `Deno.stdin.isTerminal()`.
+
+### Invalid:
+
+```typescript
+Deno.isatty(Deno.stdin.rid);
+Deno.isatty(Deno.stdout.rid);
```
Comment (iuioiua) [inline docs/rules/no_deprecated_deno_api.md:132]:
```suggestion
Deno.isatty(Deno.stdout.rid);
Deno.isatty(Deno.stderr.rid);
```

File: docs/rules/no_deprecated_deno_api.md [hunk 6]
```diff
@@ -108,3 +108,47 @@ function foo(file: Deno.FsFile) {
   // ...
 }
 ```
+
+- `Deno.fdatasync`
+- `Deno.fdatasyncSync`
+
+`Deno.fdatasync` and `Deno.fdatasyncSync` are deprecated in favor of
+`Deno.FsFile.datasync` and `Deno.FsFile.datasyncSync` respectively.
```
Comment (iuioiua) [inline docs/rules/no_deprecated_deno_api.md:116]:
```suggestion
`Deno.FsFile.flushData` and `Deno.FsFile.flushDataSync` respectively.
```

**Review by iuioiua — CHANGES_REQUESTED** [review]

File: docs/rules/no_deprecated_deno_api.md (hunk 2 above)
Comment (iuioiua) [reply docs/rules/no_deprecated_deno_api.md:122]: Ditto for other parts relating to this API.

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**